src/main/java/redis/clients/jedis/util/JedisURIHelper.java

```diff
@@ -70,6 +71,73 @@ public static RedisProtocol getRedisProtocol(URI uri) {
     return null; // null (default) when not defined
   }
 
+  private static final Integer ZERO_INTEGER = 0;
+
+  public static ClientSideCache getClientSideCache(URI uri) {
+    if (uri.getQuery() == null) return null;
+
+    boolean guava = false, caffeine = false; // cache_lib
+    Integer maxSize = null; // cache_max_size --> 0 = disbale
+    Integer ttl = null; // cache_ttl --> 0 = no ttl
+    // cache-max-idle
+
+    String[] params = uri.getQuery().split("&");
+    for (String param : params) {
+      int idx = param.indexOf("=");
```

Comment:
Check `idx`, might be that the URI parameters are broken, maybe we want to give a friendly error message.